chesster.js

```diff
@@ -698,28 +691,58 @@ function parsePairingForRound(roundSheet, requestingPlayer, targetPlayer, callba
+    var localDateTimeString = localTime.format("dddd [at] HH:mm");
+    var player = details.player;
+    var opponent = details.opponent;
+    var color = details.color;
+    var rating = details.rating;
+
+    if (!localTime.isValid()) {
+        callback(player + " will play as " + color +" against " + opponent + getRatingString(rating) + ".  The game is unscheduled.");
+    } else if (moment.utc().isAfter(localTime)) {
+        // If the match took place in the past, display the date instead of the day
+        localDateTimeString = localTime.format("MM/DD [at] HH:mm");
+        callback(player + " played as " + color +" against " + opponent + getRatingString(rating) + " on " + localDateTimeString + ".");
+    } else {
+        // Otherwise display the time until the match
+        var timeUntil = localTime.fromNow(true);
+        callback(player + " will play as " + color +" against " + opponent + getRatingString(rating) + " on " + localDateTimeString + " which is in " + timeUntil + ".");
+    }
+}
+
 function parsePairingResult(player, tz_offset, opponent, color, date, time, callback) {
 
     var opponentName = uncaptain(opponent);
-    
-    getPlayerByName(opponentName, function(opponent) {
-        getClassicalRating(opponent, function(rating) {
-            var localTime = moment.utc(date + " " + time, "MM/DD HH:mm").utcOffset(tz_offset);
-            var localDateTimeString = localTime.format("dddd [at] HH:mm");
-
-            if (!localTime.isValid()) {
-                callback(player + " will play " + opponentName + "(" + rating + ").  The game is unscheduled.");
-            } else if (moment.utc().isAfter(localTime)) {
-                // If the match took place in the past, display the date instead of the day
-                localDateTimeString = localTime.format("MM/DD [at] HH:mm");
-                callback(player + " played " + opponentName + " (" + rating + ") on " + localDateTimeString + ".");
-            } else {
-                // Otherwise display the time until the match
-                var timeUntil = localTime.fromNow(true);
-                callback(player + " will play " + opponentName + " (" + rating + ") on " + localDateTimeString + " which is in " + timeUntil + ".");
-            }
-
-        });
+    getPlayerByName(opponentName, function(opponent, error) {
+        if(opponent){
+            getClassicalRating(opponent, function(rating) {
+                formatPairingResult({
+                    "player": player, 
+                    "tz_offset": tz_offset, 
+                    "opponent": opponentName, 
+                    "color": color, 
+                    "date": date, 
+                    "time": time, 
+                    "rating": rating, 
+                }, callback);
+            });
+        }else{
+            console.error(JSON.stringify(error));
+            formatPairingResult({
```

Comment:
in the case that we cannot find a rating, it could be because the api is down. 

no reason to not display the pairing without the rating.